analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -72,6 +120,24 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
         };
     }
 
+    private static IOperation CanLearnNotNull(IInvocationOperationWrapper invocation)
+    {
+        if (invocation.Instance is not null && !invocation.TargetMethod.IsExtensionMethod && !IsNullableGetValueOrDefault(invocation))
+        {
+            return invocation.Instance;
+        }
+        else if (invocation.TargetMethod.ContainingType.Is(KnownType.System_Linq_Enumerable)
+            && LinqMethodsNullCheckingSource.Contains(invocation.TargetMethod.Name)) // The name comes from the symbol, so no need to be case-insensitive for VB.NET
```

Comment:
There's an extension for this. 
I don't feel the need to comment about VB, because it's clearly testing a Symbol, not a Syntax.
```suggestion
        else if (invocation.TargetMethod.IsAny(KnownType.System_Linq_Enumerable, LinqMethodsNullCheckingSource))
```